pkg/netdevice/netResourcePool_test.go

```diff
@@ -237,4 +237,107 @@ var _ = Describe("NetResourcePool", func() {
 			})
 		})
 	})
+	Describe("Health Checking on Net Devices", func() {
+		Context("PF link status and device existance for VFs with Same PF", func() {
+			rf := factory.NewResourceFactory("fake", "fake", true, false)
+			nadutils := rf.GetNadUtils()
+			rc := &types.ResourceConfig{
+				ResourceName:             "fake",
+				ResourcePrefix:           "fake",
+				CheckHealthOnPf:          true,
+				CheckHealthOnDeviceExist: true,
+				SelectorObjs:             []interface{}{&types.NetDeviceSelectors{}},
+			}
+
+			var fake1health bool = true
+			var fake2health bool = true
+
+			fake1 := &mocks.PciNetDevice{}
+			fake1.On("GetPfNetName").Return("fakepf1")
+			fake1.On("GetPciAddr").Return("0000:01:00.1")
+			fake1.On("SetHealth", Anything).Run(func(args Arguments) {
+				fake1health = args.Bool(0)
+			}).Return()
+
+			fake2 := &mocks.PciNetDevice{}
+			fake2.On("GetPfNetName").Return("fakepf1")
+			fake2.On("GetPciAddr").Return("0000:01:00.2")
+			fake2.On("SetHealth", Anything).Run(func(args Arguments) {
+				fake2health = args.Bool(0)
+			}).Return()
+
+			pcis := map[string]types.HostDevice{"fake1": fake1, "fake2": fake2}
+
+			rp := netdevice.NewNetResourcePool(nadutils, rc, pcis)
```

Comment:
can you move this section in a `BeforeEach()`? code outside ginkgo construct is called very early during tests.
This would also allow getting rif of the `mock.Unset()` calls

Reply:
Ah very good point, I have implemented this, it is much cleaner now, thanks!